article-store/articles/api.py

```diff
@@ -74,7 +75,7 @@ def post(self, request: Request, article_id: str) -> HttpResponse:
         with transaction.atomic():
             article, created = Article.objects.get_or_create(id=article_id)
 
-            with message_publisher('article.version.post', article_id, article.next_version):
+            with message_publisher('article.version.post', request.META.get(settings.RUN_ID_HEADER)):
```

Comment:
`request` specific to the controller layer, `message_publisher` to the infrastructural layer around the models, that fits. Later some kind of creational pattern will emerge